functions/Remove-DbaDbSynonym.ps1

```diff
@@ -0,0 +1,161 @@
+    param (
+        [parameter(ValueFromPipeline)]
+        [DbaInstance[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$Database,
+        [string[]]$ExcludeDatabase,
+        [string[]]$Schema,
+        [string[]]$ExcludeSchema,
+        [string[]]$Synonym,
+        [string[]]$ExcludeSynonym,
+        [parameter(ValueFromPipeline)]
+        [object[]]$InputObject,
+        [switch]$EnableException
+    )
+
+    process {
+        if (-not $InputObject -and -not $SqlInstance) {
+            Stop-Function -Message "You must pipe in a synonym, database, or server or specify a SqlInstance"
+            return
+        }
+
+        if ($SqlInstance) {
+            $InputObject = $SqlInstance
+        }
+
+        foreach ($input in $InputObject) {
+            $inputType = $input.GetType().FullName
+            switch ($inputType) {
+                'Sqlcollaborative.Dbatools.Parameter.DbaInstanceParameter' {
+                    Write-Message -Level Verbose -Message "Processing DbaInstanceParameter through InputObject"
+                    $dbSynonyms = Get-DbaDBSynonym -SqlInstance $input -SqlCredential $SqlCredential -Database $Database -ExcludeDatabase $ExcludeDatabase -Schema $Schema -ExcludeSchema $ExcludeSchema -Synonym $Synonym -ExcludeSynonym $ExcludeSynonym
+                }
+                'Microsoft.SqlServer.Management.Smo.Server' {
+                    Write-Message -Level Verbose -Message "Processing Server through InputObject"
+                    $dbSynonyms = Get-DbaDBSynonym -SqlInstance $input -SqlCredential $SqlCredential -Database $Database -ExcludeDatabase $ExcludeDatabase -Schema $Schema -ExcludeSchema $ExcludeSchema -Synonym $Synonym -ExcludeSynonym $ExcludeSynonym
+                }
+                'Microsoft.SqlServer.Management.Smo.Database' {
+                    Write-Message -Level Verbose -Message "Processing Database through InputObject"
+                    $dbSynonyms = $input | Get-DbaDBSynonym -ExcludeDatabase $ExcludeDatabase -Schema $Schema -ExcludeSchema $ExcludeSchema -Synonym $Synonym -ExcludeSynonym $ExcludeSynonym
+                }
+                'Microsoft.SqlServer.Management.Smo.Synonym' {
+                    Write-Message -Level Verbose -Message "Processing DatabaseSynonym through InputObject"
+                    $dbSynonyms = $input
+                }
+                default {
+                    Stop-Function -Message "InputObject is not a server, database, or database synonym."
+                    return
+                }
+            }
+
+            foreach ($dbSynonym in $dbSynonyms) {
+                $db = $dbSynonym.Parent
+                $instance = $db.Parent
+
+                if ($PSCmdlet.ShouldProcess($instance, "Remove synonym $dbSynonym from database $db")) {
+
+                    try {
+                        # avoid enumeration issues
+                        $db.Query("DROP SYNONYM $dbSynonym")
```

Comment:
Does the SMO class not support dropping this object type?

Reply:
It does, but when I tried to repeat the behaviour of Remove-DbaDbRole  I've got the enumeration issues, so I've used a workaround from a different function.

Reply:
Cannot figure out how to make it. 
`Remove-DbaDbSynonym -SqlInstance $s1 -Database Mikey -Synonym syn2, syn3` works fine, but when I do
`Get-DbaDbSynonym -SqlInstance $s1 -Database Mikey | Remove-DbaDbSynonym` 
it returns an error
```Collection was modified; enumeration operation may not execute.
At C:\Users\micha\OneDrive\learn\dbatools\functions\Get-DbaDbSynonym.ps1:140 char:22
+             foreach ($dbSynonym in $dbSynonyms) {
```

I've tried different ways (used in on other commands in dbatools, but none seems to be as effective as using plain T-SQL)